### PR TITLE
Run the virtiofsd container as non-root for config volumes

### PR DESCRIPTION
**Pull request by germag**
**What this PR does / why we need it**:

Currently, we can share configuration volumes, such as ConfigMaps, Secrets, DownwardAPI and ServiceAccount using virtiofs, but the virtiofs container requires to be run as root. 

Virtiofsd supports being run without privileges, so this PR to modify the container to run as the unprivileged user when sharing configuration volumes.

Reported-by: Javier Cano Cano <jcanocan@redhat.com>

Tested-by: Javier Cano Cano <jcanocan@redhat.com>

Related #7735

**Release note**:

```release-note
Add support for running virtiofsd in an unprivileged container when sharing configuration volumes.
```

### CODE REVIEW

**Comment by germag** [comment]
v2:
- Add AllowPrivilegeEscalation field to SecurityContext

**Comment by alicefr** [comment]
@germag I have a question. IIUC, we are adding the noroot version but still keeping the root configuration option available. However, in the test now we are only testing noroot, do I understand the change correctly?

**Comment by germag** [comment]
> @germag I have a question. IIUC, we are adding the noroot version but still keeping the root configuration option available. However, in the test now we are only testing noroot, do I understand the change correctly?

Yes, but we only keep the root configuration for PVCs(*), and we keep testing those as root `tests/virtiofs/datavolume.go`. The non-root (for now) it's only for config volumes, but you cannot choose sharing config volumes as root for that reason we only test those as non-root.

(*) Currently, we can share PVC as non-root, but the functionality is very limited.

**Comment by germag** [comment]
> We should onboard you to community.

Thanks, should I need to do something?

**Comment by jcanocan** [comment]
Looking awesome for me! Please, add a release note.

**Comment by alicefr** [comment]
/approve
/hold
If anyone else want to review it.

**Review by xpivarc — COMMENTED** [review]
LGTM

File: pkg/virt-controller/services/virtiofs.go [hunk 1]
```diff
@@ -71,33 +71,67 @@ func resourcesForVirtioFSContainer(dedicatedCPUs bool, guaranteedQOS bool, confi
...
-		"SETFCAP",
-		"SYS_CHROOT",
+type securityProfile uint8
+
+const (
+	restricted securityProfile = iota
+	privileged
+)
+
+func isRestricted(profile securityProfile) bool {
```
[the remark this answers is not in the document]
Comment (germag) [reply pkg/virt-controller/services/virtiofs.go:83]: Do you mean, replacing it by just `profile == restricted`?, I did it first, but it felt less clear, specially in:
```go
		RunAsNonRoot:             pointer.Bool(profile == restricted),
		AllowPrivilegeEscalation: pointer.Bool(profile == privileged),
```

but I don't have a strong opinion about that

**Pull request merged**